iml-api/src/graphql.rs

```diff
@@ -357,46 +359,81 @@ impl QueryRoot {
                     message
                 FROM chroma_core_command c
                 JOIN chroma_core_command_jobs cj ON c.id = cj.command_id
-                WHERE complete = $4
+                WHERE ($4::BOOL IS NULL OR complete = $4)
                   AND ($5::TEXT IS NULL OR c.message ILIKE '%' || $5 || '%')
                 GROUP BY c.id
                 ORDER BY
                     CASE WHEN $3 = 'asc' THEN c.id END ASC,
                     CASE WHEN $3 = 'desc' THEN c.id END DESC
-                OFFSET $1 LIMIT $2 "#,
+                OFFSET $1 LIMIT $2
+            "#,
             offset.unwrap_or(0) as i64,
             limit.unwrap_or(20) as i64,
             dir.deref(),
             is_completed,
             msg,
         )
         .fetch_all(&context.pg_pool)
-        .map_ok(|xs: Vec<_>| {
-            xs.into_iter()
-                .map(|x| Command {
-                    id: x.id,
-                    cancelled: x.cancelled,
-                    complete: x.complete,
-                    errored: x.errored,
-                    created_at: x.created_at.format("%Y-%m-%dT%T%.6f").to_string(),
-                    jobs: {
-                        x.job_ids
-                            .unwrap_or_default()
-                            .into_iter()
-                            .map(|job_id: i32| {
-                                format!("/api/{}/{}/", Job::<()>::endpoint_name(), job_id)
-                            })
-                            .collect::<Vec<_>>()
-                    },
-                    logs: "".to_string(),
-                    message: x.message.clone(),
-                    resource_uri: format!("/api/{}/{}/", Command::endpoint_name(), x.id),
-                })
-                .collect::<Vec<_>>()
+        .map_ok(|xs: Vec<CommandTmpRecord>| {
+            xs.into_iter().map(to_command).collect::<Vec<Command>>()
         })
         .await?;
         Ok(commands)
     }
+
+    /// Fetch the list of commands by ids, the returned
+    /// collection is guaranteed to match the input.
+    /// If a command not found, `None` is returned for that index.
+    #[graphql(arguments(
+        limit(description = "paging limit, defaults to 20"),
+        offset(description = "Offset into items, defaults to 0"),
+        ids(description = "The list of command ids to fetch, ids may be empty"),
+    ))]
+    async fn commands_by_ids(
+        context: &Context,
+        limit: Option<i32>,
+        offset: Option<i32>,
+        ids: Vec<i32>,
+    ) -> juniper::FieldResult<Vec<Option<Command>>> {
```

Comment:
nit:

If we pass in an invalid id, it might be better to just return an error. With this approach, we will have to `.filter_map` every result we get back from this call.

Reply:
I'll tackle it in follow up PR.